plasmapy/atomic/ionization_states.py

```diff
@@ -0,0 +1,87 @@
+import numpy as np
+from .particle_class import Particle
+from .particle_input import particle_input
+from ..utils import AtomicError
+
+class IonizationState:
+    """
+    Describe the ionization state of a single element.
+    """
+    @particle_input(require='element', exclude=['isotope', 'ion'])
+    def __init__(self,
+                 element: Particle,
+                 distribution=None,
+                 T_e=None,  # not yet implemented!
+                 ):
+        """
+        Initialize the ionization state distribution.
+
+        Notes
+        -----
+        The method to calculate collisional ionization equilibrium has
+        not yet been implemented and will raise a `NotImplementedError`.
+
+        """
+        self._element = element
+
+        if distribution is None:
+            raise NotImplementedError(
+                "IonizationState instances are not yet able to "
+                "calculate collisional ionization equilibrium.")
+
+        if isinstance(distribution, (tuple, list)):
+            distribution = np.array(distribution)
+        elif not isinstance(distribution, np.ndarray):
+            raise TypeError
+
+        if len(distribution) != element.atomic_number + 1:
+            raise AtomicError("Incorrect number of ionization states.")
+
+        ionsum = np.sum(distribution)
+        if not np.isclose(ionsum, 1.0):
+            raise ValueError(
+                f"The sum of the ionization states of {self.element} "
+                f"is {ionsum}, which does not approximately equal one.")
+
+        self._ionization_state = distribution
+
+    def __getitem__(self, value):
```

Comment:
add `__iter__` so we can iterate through charge states.

Reply:
Probably not going to be trivial, especially on the design front. But it's an idea.

Reply:
I've never implemented `__iter__` before, so that sounds really fun!  It will probably also be really useful.

It's also kind of funny to have a magic method in a plasma package called `__iter__`!  ...though I suppose it would be funnier if it were `__ITER__`.

Reply:
also worth noting, you'll want to implement `__next__` along with `__iter__`

Reply:
Thanks, I didn't know that!  I would have guessed `__DEMO__`.

Reply:
I'm wondering how best to do this iteration, since we might want to get any combination of the integer charge, the `Particle` instance, the ionic symbol, and the ionization fraction for that state.  Maybe a `collections.namedtuple`?  Something like...

```Python
for state in ionization_states:  # using maximally verbose attribute nomenclature
    print(state.integer_charge, state.ionization_fraction, state.particle_instance)
```

Reply:
I think a tuple is reasonable, especially since then you could do:
```Python
for charge, fraction, particle in ionizationInfo:
    # important physics here
```